src/core/rtkQuery/splitV3Api/userApi.ts

```diff
@@ -30,54 +30,53 @@ const userApi = splitV3Api.injectEndpoints({
       providesTags: ['Users'],
     }),
 
-    // Edit User. This replaces all values, except Plex and Password.
-    putUser: build.mutation<void, UserType>({
+    // Add a new user.
+    postUser: build.mutation<UserType, Omit<UserType, 'ID'>>({
       query: ({ CommunitySites, ...user }) => ({
         url: 'User',
-        method: 'PUT',
-        body: { ...user, CommunitySites: map(pickBy(CommunitySites, identity), (_, key) => key) },
+        method: 'POST',
+        body: { ...user, CommunitySites: CommunitySites ? map(pickBy(CommunitySites, identity), (_, key) => key) : null } as ApiUserType,
       }),
+      transformResponse: (user: ApiUserType) => {
+        const { CommunitySites, ...tempUser } = user;
+        return { ...tempUser, CommunitySites: simplifyCommunitySites(CommunitySites) };
+      },
       invalidatesTags: ['Users'],
     }),
 
-    // Change the password for a user
-    postChangePassword: build.mutation<void, { Password: string, RevokeAPIKeys: boolean, userId: number, admin: boolean }>({
-      query: ({ admin, userId, ...body }) => ({
-        url: `User/${admin ? userId.toString() : 'Current'}/ChangePassword`,
-        body,
-        method: 'POST',
+    // Edit the current user or a user by id using a raw object to do a partial update.
+    putUser: build.mutation<UserType, UserType>({
+      query: ({ ID, CommunitySites, Password: __, ...user }) => ({
```

Comment:
I made sure to omit the Password from the body to send so the server won't complain — even if we accidentally include it at some point — since our user type (`UserType`) allows it to included, but the server will reject the property since it's an unknown property from it's point of view. This is mostly because our type don't fully match what the server is expecting, but if you two are fine with it then I won't complain.